### PR TITLE
Xa12850432

### PR DESCRIPTION
**Pull request by AlexandrCherednyk**
Lab2

### CODE REVIEW

File: Lab2/rock-paper-scissors.c [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    else if (result == 1) {
+        printf("\n\n\t\t\t\tWow! You have won the game!\n");
+    }
+    else {
+        printf("\n\n\t\t\t\tOh! You have lost the game!\n");
+    }
+    printf("\t\t\t\tYOu choose : %c and Computer choose : %c\n", you, computer);
+
+    return 0;
+}
```
Comment (MarynaMalakhovaKarazinUA) [inline Lab2/rock-paper-scissors.c:68]: no new line at the end of file

File: Lab2/rock-paper-scissors.c [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    if (result == -1) {
+        printf("\n\n\t\t\t\tGame Draw!\n");
+    }
+    else if (result == 1) {
+        printf("\n\n\t\t\t\tWow! You have won the game!\n");
+    }
+    else {
+        printf("\n\n\t\t\t\tOh! You have lost the game!\n");
+    }
+    printf("\t\t\t\tYOu choose : %c and Computer choose : %c\n", you, computer);
```
Comment (MarynaMalakhovaKarazinUA) [inline Lab2/rock-paper-scissors.c:65]: o

File: Lab2/rock-paper-scissors.c (hunk 1 above)
Comment (MarynaMalakhovaKarazinUA) [reply Lab2/rock-paper-scissors.c:68]: Nice job! But you have some problems with the commit name (tusk) and a few comments, please, fix it :)))

File: Lab3/A.c [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		}
+
+	}
+	puts(stringPointer);
+	fputs(stringPointer, fp);
+	fclose(fp);
+	free(stringPointer);
+
+	return 0;
+}
```
Comment (MarynaMalakhovaKarazinUA) [inline Lab3/A.c:53]: no line at the end

File: Lab3/B.c [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    }
+    printf("\t%c: %d\n", string[externalCounter], character);
+
+    while (string[externalCounter + 1] == string[externalCounter]) {
+      externalCounter++;
+    }
+  }
+  fclose(fp);
+  return 0;
+}
```
Comment (MarynaMalakhovaKarazinUA) [inline Lab3/B.c:32]: no line at the end

File: Lab3/makefile.mk [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+	gcc -c A.c -o A.o
+
+A.exe: A.o
+	gcc A.o -o A.exe
+
+B.o: B.c
+	gcc -c B.c -o B.o
+
+B.exe: B.o
+	gcc B.o -o B.exe
```
Comment (MarynaMalakhovaKarazinUA) [inline Lab3/makefile.mk:13]: no new line

Comment (MarynaMalakhovaKarazinUA) [reply Lab3/makefile.mk:13]: There is incorrect commits` name

File: Lab2/rock-paper-scissors.c (hunk 1 above)
Comment (MarynaMalakhovaKarazinUA) [reply Lab2/rock-paper-scissors.c:68]: Why do you need the *.mk extension at the end of Makefile name?

File: Lab3/makefile.mk (hunk 5 above)
Comment (MarynaMalakhovaKarazinUA) [reply Lab3/makefile.mk:13]: You have a strange Makefile name, how do you use it?

**Comment by AlexandrCherednyk** [comment]
Thanks for your comments, i will fix the bugs shortly.

пт, 12 нояб. 2021 г. в 10:54, MarynaMalakhovaKarazinUA <
***@***.***>:

File: Script.txt [hunk 6]
```diff
@@ -0,0 +1,2 @@
+$ find /e/lab2_Cherednyk -type f -name "S*"
```
Comment (MarynaMalakhovaKarazinUA) [inline Script.txt]: It is not a script...

File: Script.sh [hunk 7]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+cd $1
+ls -la | grep "$2" >>  /e/lab2_Cherednyk/answer_list.txt
```
Comment (MarynaMalakhovaKarazinUA) [inline Script.sh]: 1. there is no new line at the end of the file.
2. it is weird that you use such directory... For example, I have no it, maybe it should be added

File: lab6/3_5_5.c [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    chdir("/"); // changing to root dir
+
+
+    if (setsid() < 0) // setting new session id
+
+        exit(0);
+
+
+    printf("%d\n", getpid());
+
```
Comment (MarynaMalakhovaKarazinUA) [inline lab6/3_5_5.c:30]: extra lines

Comment (MarynaMalakhovaKarazinUA) [reply lab6/3_5_5.c:30]: Also your commit name is weird. I seems to be something like: "lab6: Add solution" and some message. You do not need comma

**Pull request merged**